spec/requests/api/moves_controller_index_v2_spec.rb

```diff
@@ -0,0 +1,235 @@
+        }
+      end
+      let(:params) { { filter: filters } }
+      let(:expected_move) do
+        {
+          data: [
+            {
+              id: move.id,
+              attributes: {
+                cancellation_reason: move.cancellation_reason,
+                cancellation_reason_comment: move.cancellation_reason_comment,
+              },
+            },
+          ],
+        }
+      end
+
+      it 'returns the correct attributes values for moves' do
+        do_get
+
+        expect(response_json).to include_json(expected_move)
+      end
+    end
+
+    context 'with a booked move' do
+      let(:move) { create(:move, :booked) }
+      let!(:moves) { [move] }
+      let(:from_location_id) { move.from_location_id }
+      let(:filters) do
+        {
+          from_location_id: from_location_id,
+        }
+      end
+      let(:params) { { filter: filters } }
+
+      it 'returns the correct attributes values for moves' do
+        do_get
+        expect(response_json).to include_json(
+          data: [{ id: move.id }],
+        )
+      end
+    end
+
+    describe 'paginating results' do
+      let!(:moves) { create_list :move, 6 }
+
+      let(:meta_pagination) do
+        {
+          per_page: 5,
+          total_pages: 2,
+          total_objects: 6,
+          links: {
+            first: '/api/moves?page=1',
+            last: '/api/moves?page=2',
+            next: '/api/moves?page=2',
+          },
+        }
+      end
+
+      before { do_get }
+
+      it_behaves_like 'an endpoint that paginates resources'
+    end
+
+    describe 'validating dates before running queries' do
+      let(:from_location) { moves.first.from_location }
+      let(:filters) do
+        {
+          from_location_id: from_location.id,
+          date_from: 'yyyy-09-Tu',
+        }
+      end
+      let(:params) { { filter: filters } }
+
+      before { do_get }
+
+      it_behaves_like 'an endpoint that responds with error 422' do
+        let(:errors_422) do
+          [{ 'title' => 'Invalid date_from',
+             'detail' => 'Validation failed: Date from is not a valid date.' }]
+        end
+      end
+    end
+
+    describe 'included relationships' do
+      let!(:moves) do
+        create_list(
+          :move,
+          1,
+          from_location: from_location,
+          to_location: to_location,
+        )
+      end
+
+      let!(:court_hearing) { create(:court_hearing, move: moves.first) }
+
+      let(:to_location) { create(:location, suppliers: [supplier]) }
+      let(:from_location) { create(:location, suppliers: [supplier]) }
+
+      before do
+        get "/api/moves#{query_params}", params: params, headers: headers
+      end
+
+      context 'when not including the include query param' do
+        let(:query_params) { '' }
+
+        it 'returns the default includes' do
+          returned_types = response_json['included']
+          expect(returned_types).to be_nil
```

Comment:
not sure, maybe I am missing something, where is the action in this test?

Reply:
Line 172

Reply:
Unless I'm misunderstanding?